packages/mesos/extra/dcos-drain-mesos-agent-public.service

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Drain Mesos Agent: Stop Mesos tasks on shutdown/reboot
+Before=shutdown.target reboot.target halt.target
+After=dcos-mesos-slave-public.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+ExecStop=/usr/bin/systemctl kill -s SIGUSR1 dcos-mesos-slave-public.service
```

Comment:
I did not check if the agent ID persisted after the reboot.  I think it did register as the same agent. I can explicitly test that.

For behaviour of Marathon, I noticed in my test that the drained tasks were attempted to be replaced right away. 

Right now Marathon does have default unreachable strategy to replace unreachable task  immediately. But it does take sometime for a task to be marked as Unreachable after the agent is shutdown (depends on the agent ping timeout if I remember correctly). So the behaviour would be that if the nodes are drained the task would be replace more promptly than if it were not drained. 
Data services would try to wait for the agent to come back online to re-spin the task back on the same node (unless a human intervenes) 

Definitely if Mesos did a TASK_KILLED instead of TASK_LOST it would be awesome.